record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java

```diff
@@ -560,63 +567,83 @@ private CodeBlock buildCodeBlock() {
         return codeBuilder.build();
     }
 
-    private void addStaticFromWithMethod() {
+    private TypeName buildWithTypeName()
+    {
+        ClassName rawTypeName = ClassName.get(packageName, builderClassType.name() + "." + metaData.withClassName());
+        if (typeVariables.isEmpty()) {
+            return rawTypeName;
+        }
+        return ParameterizedTypeName.get(rawTypeName, typeVariables.toArray(new TypeName[]{}));
+    }
+
+    private void addFromWithClass() {
         /*
-            Adds static method that returns a "with"er view of an existing record.
+            Adds static private class that implements/proxies the Wither
 
-            public static With from(MyRecord from) {
-                return new MyRecordBuilder.With() {
-                    @Override
-                    public String p1() {
-                        return from.p1();
-                    }
-
-                    @Override
-                    public String p2() {
-                        return from.p2();
-                    }
-                };
+            private static final class _FromWith implements MyRecordBuilder.With {
```

Comment:
Class name will be `MyRecordBuilder_FromWith` ?

Reply:
`MyRecordBuilder._FromWith` - The `_` is commonly used to hide internal names. But, I'm open to other names.

Reply:
`_FromWith` is good, I just did not read the code properly